micro-rdk/src/common/conn/viam.rs

```diff
@@ -538,6 +538,34 @@ where
         ));
         self.app_client_tasks.push(config_monitor_task);
 
+        #[cfg(feature = "ota")]
+        {
+            use crate::esp32::ota;
+
+            log::debug!("ota feature enabled");
+
+            if let Some(service) = config
+                .services
+                .iter()
+                .find(|&service| service.model == ota::OTA_MODEL_TRIPLET)
+            {
+                log::debug!("service config: {:#?}", service);
+                if let Ok(mut ota) = ota::OtaService::from_config(&service) {
+                    log::debug!("OtaService: {:?}", ota);
+                    if let Err(e) = ota.update().await {
```

Comment:
Yes. This doesn't include the retry/backoff logic yet. 
I wanted to discuss further if we should move this to a PeriodicTask. 
Otherwise we would need to block here or spawn a thread to handle retries.

What do you suggest?

Reply:
retry logic to be handled with exponential backoff ticket in description